bundles/org.openhab.binding.ecovacs/src/main/java/org/openhab/binding/ecovacs/internal/handler/EcovacsVacuumHandler.java

```diff
@@ -0,0 +1,830 @@
+                device.sendCommand(new SetVolumeCommand((volumePercent + 5) / 10));
+                return;
+            } else if (channel.equals(CHANNEL_ID_SUCTION_POWER) && command instanceof StringType) {
+                Optional<SuctionPower> power = SUCTION_POWER_MAPPING.findMappedEnumValue(command.toString());
+                if (power.isPresent()) {
+                    device.sendCommand(new SetSuctionPowerCommand(power.get()));
+                    return;
+                }
+            } else if (channel.equals(CHANNEL_ID_WATER_AMOUNT) && command instanceof StringType) {
+                Optional<MoppingWaterAmount> amount = WATER_AMOUNT_MAPPING.findMappedEnumValue(command.toString());
+                if (amount.isPresent()) {
+                    device.sendCommand(new SetMoppingWaterAmountCommand(amount.get()));
+                    return;
+                }
+            } else if (channel.equals(CHANNEL_ID_AUTO_EMPTY)) {
+                if (command instanceof OnOffType) {
+                    device.sendCommand(new SetDustbinAutoEmptyCommand(command == OnOffType.ON));
+                    return;
+                } else if (command instanceof StringType && command.toString().equals("trigger")) {
+                    device.sendCommand(new EmptyDustbinCommand());
+                    return;
+                }
+            } else if (channel.equals(CHANNEL_ID_TRUE_DETECT_3D) && command instanceof OnOffType) {
+                device.sendCommand(new SetTrueDetectCommand(command == OnOffType.ON));
+                return;
+            } else if (channel.equals(CHANNEL_ID_CONTINUOUS_CLEANING) && command instanceof OnOffType) {
+                device.sendCommand(new SetContinuousCleaningCommand(command == OnOffType.ON));
+                return;
+            } else if (channel.equals(CHANNEL_ID_CLEANING_PASSES) && command instanceof DecimalType) {
+                int passes = ((DecimalType) command).intValue();
+                device.sendCommand(new SetDefaultCleanPassesCommand(passes));
+                lastDefaultCleaningPasses = passes; // if we get here, the command was executed successfully
+                return;
+            }
+            logger.debug("{}: Ignoring unsupported device command {} for channel {}", getDeviceSerial(), command,
+                    channel);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (EcovacsApiException e) {
+            logger.debug("{}: Handling device command {} failed", getDeviceSerial(), command, e);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final String serial = getConfigAs(EcovacsVacuumConfiguration.class).serialNumber;
+        if (serial.isEmpty()) {
+            logger.info("Thing {} is missing serial number information", getThing().getUID());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
```

Comment:
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/offline-conf-error-missing-serial-number");
```

(where `offline-conf-error-missing-serial-number` refers to an I18N string)